<!doctype html>
<html lang="en">
<head>
	<meta charset="utf-8">
	<title>My Research | Yin Jun, Phua -- Assistant Professor at Tokyo Tech</title>
	<meta name="description" content="Yin Jun Phua, currently an assistant professor at Tokyo Institute of Technology. Main research focus is on brdging the gap between symbolic AI and neural networks.">
	<link rel="stylesheet" href="/css/chota.css">
	<link rel="stylesheet" href="/css/style.css">
	<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
	<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
	<link rel="manifest" href="/site.webmanifest">
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<link rel="alternate" hreflang="en" href="https://yinjunphua.com/my-research.html">
	<link rel="alternate" hreflang="ja" href="https://yinjunphua.com/ja/my-research.html">
	<link rel="canonical" href="https://yinjunphua.com/my-research.html">
	<meta http-equiv="content-language" content="en">
</head>
<body>
	<div class="container">
		<div class="row">
			<div class="col">
				<div class="pull-right">
					EN | <a href="/ja/my-research.html">日本語</a>
				</div>
			</div>
		</div>
		<nav class="nav">
			<div class="nav-left">
				<a class="is-paddingless" href="/"><h2 class="is-marginless">Yin Jun, Phua <small>&mdash; My Research</small></h2></a>
			</div>
			<div class="nav-right">
				<a href="/my-research.html">My Research</a>
			</div>
		</nav>
		<hr>
		<h2>Learning Logic Programs from Dynamic Systems</h2>
		<div class="row">
			<div class="col-5">
				<picture>
					<source srcset="/images/ilp-model-2017.webp" type="image/webp">
					<source srcset="/images/ilp-model-2017.png" type="image/png">
					<img src="/images/ilp-model-2017.png" alt="Learning to classify logic programs by time series data">
				</picture>
			</div>
			<div class="col-2"></div>
			<div class="col-5">
				<picture>
					<source srcset="/images/delta-lfit-model-2019.webp" type="image/webp">
					<source srcset="/images/delta-lfit-model-2019.png" type="image/png">
					<img src="/images/delta-lfit-model-2019.png" alt="Learning logical rules from time series data">
				</picture>
			</div>
		</div>
		<p>My main research focus has been on a neuro-symbolic method to learn transition rules from time series data. Using a singular neural network to learn to classify different time series that originate from different dynamic systems, we have shown that it was possible to predict the next state from just the historical data (Phua, et al. <i>27th International Conference on Inductive Logic Programming</i>, 2017). By inputting the time series into an LSTM (Long-Short Term Memory) and using the corresponding logic program as the hidden states, the LSTM was able to learn representations of the logic program and used the representation to predict the next state. Next, we have also shown that neural network can identify different time series that come from the same dynamic system, and we also revised our method to take into account background knowledge (Phua, et al. <i>IfCoLog Journal of Applied Logics</i>, 2019). By training the LSTM to identify different time series from the same system, we were able to improve the accuracy of our method.</p>
		<p>Using the above results, and leveraging the strength of neural networks for pattern recognition, we proposed a method that learns logical rules from state transition data (Phua, et al. 29th International Conference on Inductive Logic Programming, 2019). By classifying the rules that are inherent within the state transitions, it becomes possible to output a logic program that represents the dynamic system. To train the neural network to classify all possible rules, we set some constraint on the learned rules. By not touching the architecture of the neural network itself, but levaraging the neural network for classification of logical rules, we were able to prove that the resulting method is robust again noise. This work was recognized by the community, and we received the Best Student Paper award. Next, by utilizing some of the invariant properties that are inherent in logic systems, we managed to surpress the combinatorial explosion problem that comes from trying to classify all possible rules (Phua, et al. <i>1st International Joint Conference on Learning &amp; Reasoning</i>, 2021). In particular, we look at the invariant in the order of the state transition input, and we proposed a neural network architecture that takes the invariance into consideration. In addition to that, by considering multiple logical rules in one inference, we were able to reduce the memory requirements of the neural network. Therefore, we were able to apply our method to larger dynamic systems, and it brings us a step closer to real world applications.</p>
		<p>We have also extended our method to delayed dynamic systems, and have also shown robustness against missing data. We were also able to extend our method to asynchronous semantics, instead of the synchronous semantics that all of our previous method considered.</p>
		<h2>Class-Incremental Learning using Generative Models</h2>
		<div class="row">
			<div class="col-8" style="margin: auto">
				<picture>
					<source srcset="/images/sddr-2023.webp" type="image/webp">
					<source srcset="/images/sddr-2023.png" type="image/png">
					<img src="/images/sddr-2023.png" alt="Levaraging generative models like Stable Diffusion to perform class-incremental learning">
				</picture>
			</div>
			<p>Statistical machine learning methods like neural networks are known to suffer from catastrophic forgetting. Given a model that has already been trained on a task, if we train the model on a new task, all knowledge regarding the previous task is lost. The field of studying this phenomenon and attempts for mitigating this, is called incremental learning. One part of it considers the task of classification, where at each training step, only a limited number of classes are available. Many different methods have been proposed with various degrees of success, and one method we consider is the method utilizing distillation and replay memory. Knowledge distillation is a method to train a model to retain knowledge from another model. Whereas replay memory is having a fixed amount of memory for previous classes. Previous methods have used instances from the training dataset to retain them as memory. However, as only a limited amount of memory is available, we can only retain a fixed amount of instances. This limits the performance of such method. In this work, we proposed a method that utilizes widely available generative models and replace them as the replay memory. We devised a method to prompt the generative model to obtain relevant images. By doing this, we were able to improve existing methods by 1~3%. We also found that there is a sweet spot between the final accuracy and the amount of images that we obtain from the generative models.</p>
			<p>Results of this work has been published in the workshop paper (Jodelet, et al. <i>1st Workshop on Visual Continual Learning</i>, 2023) and has been chosen for the best paper award.</p>
		</div>
		<h2>Consistent Identification of Gene Sets from Cell Sequencing Data</h2>
		<div class="row">
			<div class="col-4">
				<picture>
					<source srcset="/images/resvae-ensemble-2023.webp" type="image/webp">
					<source srcset="/images/resvae-ensemble-2023.png" type="image/png">
					<img src="/images/resvae-ensemble-2023.png" alt="resVAE ensemble: identifying crucial gene sets consistently from cell sequencing data">
				</picture>
			</div>
			<div class="col-8">
				<p>In collaboration with Berlin Institute of Health, we proposed a method to identify important gene sets based on the cell types from single cell sequencing data. Biologists face time pressure in analyzing cell sequencing data. Identifying gene sets quickly can help the efficiency of the development of important drugs. Previous methods only focused on the possibility of extracting gene sets, but due to randomness in statistical machine learning methods, each run potentially produces different gene sets, complicating the analysis for biologists. By utilizing an ensemble of resVAEs, we were able to stabilize the results of identified gene-sets. Further, we also show that the choice of activation function has huge implications on the stability of the results, thus we chose an activation function that provides even greater stability.</p>
				<p>Results of this work were published in the journal paper (Ten, et al. <i>Frontiers in Cell and Developmental Biology 11</i>, 2023).</p>
			</div>
		</div>
		<hr />
		<p class="text-grey text-center">Have questions or want to know more about my research work? You can always contact me by email.</p>
		<p><a href="/">&laquo; Back to home</a></p>
	</div>

	<footer class="text-center">
		<p><a href="https://orcid.org/0000-0003-1178-8238" target="_blank">ORCID</a> &bull; <a href="https://researchmap.jp/phuayj?lang=en" target="_blank">researchmap</a> &bull; <a href="https://dblp.org/pid/217/3830.html" target="_blank">dblp</a></p>
		<p>You can contact me at phua<span class="replace"> [at] </span>c.titech.ac.jp</p>
		<h5>&copy; Yin Jun Phua</h5>
	</footer>

	<script type="text/javascript">
		var selEl = (q) => document.querySelectorAll(q);
		document.addEventListener('DOMContentLoaded', function(e) {
			selEl('.replace').forEach(el => {
				el.innerHTML = '&#x40;';
			});
		});
	</script>
</body>
</html>